### PR TITLE
New methods for TaylorModel1 arithmetic

### PR DESCRIPTION
**Pull request by UzielLinares**
This PR adds new methods for TaylorModel1(with TaylorN coefficients) arithmetic.

### CODE REVIEW

File: src/arithmetic.jl [hunk 1]
```diff
@@ -129,6 +128,16 @@ function remainder_product(a, b, aux, Δnegl)
     Δ = Δnegl + Δb * a.rem + Δa * b.rem + a.rem * b.rem
     return Δ
 end
+function remainder_product(a::TaylorModel1{TaylorN{T}, S},
+                           b::TaylorModel1{TaylorN{T}, S},
+                           auxT, Δnegl) where {T, S}
+    N = get_numvars()    
+    auxQ = IntervalBox(-1 .. 1, Val(N))
```
Comment (lbenet) [inline src/arithmetic.jl:135]: Could you add somewhere a comment indicating that this method assumes that the domain (for the `TaylorN`  variables) is the symmetric normalized box?

**Comment by UzielLinares** [comment]
Hi, just pushed a test for the multiplication.

Playing around with these TM1's with TaylorN coefficients, I've found that the following needs an appropriate implementation for this kind of Taylor Models:

- Trigonometric functions (`sin(x)`, `cos(x)`, etc.)
- Exponential and logarithmic functions.
- Power functions (up to now only ^2 is supported)
 
Maybe those cases can be addressed on another PR in order to use the implemented on #84 since that PR relies on the product of TM1's with TaylorN coefficients.

File: test/TM1.jl [hunk 2]
```diff
@@ -20,6 +20,9 @@ function check_containment(ftest, tma::T) where {T<:Union{TaylorModel1, RTaylorM
     return bb
 end
 
+interval_rand(X::Interval{T}) where {T} = X.lo + rand(T) * (X.hi - X.lo)
+interval_rand(X::IntervalBox) = interval_rand.(X)
```
Comment (lbenet) [inline test/TM1.jl:24]: I think the methods of `interval_rand` are exactly equivalent to `rand(::Interval)` and `rand(::IntervalBox)`. If so, I think it is better to use directly those methods

File: src/arithmetic.jl (hunk 1 above)
Comment (lbenet) [reply src/arithmetic.jl:135]: Bump...

**Comment by lbenet** [comment]
> Hi, just pushed a test for the multiplication.
> 
> Playing around with these TM1's with TaylorN coefficients, I've found that the following needs an appropriate implementation for this kind of Taylor Models:
> 
>     * Trigonometric functions (`sin(x)`, `cos(x)`, etc.)
> 
>     * Exponential and logarithmic functions.
> 
>     * Power functions (up to now only ^2 is supported)
> 
> 
> Maybe those cases can be addressed on another PR and merge this since #84 uses the product of TM's implemented here.

Good catch! In order not to forget this, can you open an issue?

File: test/TM1.jl (hunk 2 above)
Comment (UzielLinares) [reply test/TM1.jl]: Thanks for the suggestion, I've replaced these functions with `rand`. I think it Is ready.

File: src/arithmetic.jl (hunk 1 above)
Comment (lbenet) [reply src/arithmetic.jl:135]: Can you add a comment pointing out that the domain of  validity assumed for the `TaylorN` is the  symmetric box?

**Review by lbenet — COMMENTED** [review]
I just left a comment to not forget that we assume that the TaylorN are  assumed to have the symmetric box (of proper dimension) as domain. Aside from that, I  think it is indeed ready.

**Comment by UzielLinares** [comment]
Ready!

**Review by lbenet — APPROVED** [review]
Thanks a lot!

**Pull request merged**